src/Command/Pull/PullCommandBase.php

```diff
@@ -486,10 +486,10 @@ private function importDatabaseDump(string $localDumpFilepath, string $dbHost, s
             'mysql',
         ]);
         if ($this->localMachineHelper->commandExists('pv')) {
-            $command = "pv $localDumpFilepath --bytes --rate | gunzip | MYSQL_PWD=$dbPassword mysql --host=$dbHost --user=$dbUser $dbName";
+            $command = "pv $localDumpFilepath --bytes --rate | gunzip | MYSQL_PWD=" . escapeshellarg($dbPassword) . " mysql --host=$dbHost --user=$dbUser $dbName";
```

Comment:
Instead of escaping individual arguments, you should provide arguments as environment variables and allow Symfony to escape them natively for each operating system: https://symfony.com/doc/6.4/components/process.html#using-features-from-the-os-shell

Specifically, like this example:
```php
// works the same on Windows , Linux and macOS
$process = Process::fromShellCommandline('echo "${:MESSAGE}"');
```

LocalMachineHelper::executeFromCommand() already takes env vars as an argument but you'll need to modify LocalMachineHelper::executeProcess() to actually accept those vars and pass them on to Process::start()